plumbing/bootstrap.py

```diff
@@ -74,88 +67,79 @@ def clean_error_message(message):
     return message
 
 
-def create_datapackage(description_file):
+def create_datapackage(description_file, source_folder):
     """Convert description.source.yaml into datapackage.json."""
 
-    datapackage_file = description_file.replace('yaml', 'json')
+    datapackage_file = os.path.join(source_folder, DATAPACKAGE_FILE)
 
     with open(description_file) as stream:
         try:
             package = yaml.load(stream.read())
             package['name'] = slugify(package['title'], separator='_')
             package['extraction_processors'] = select_processors(package)
-        except ParserError as error:
+        except (ParserError, ScannerError) as error:
             message = clean_error_message(str(error))
             package = {'yaml_error': message}
             logging.debug('Parser error in %s: %s', description_file, message)
 
     with open(datapackage_file, 'w+') as stream:
         stream.write(json.dumps(package, indent=4))
 
-    logging.debug('Created %s', datapackage_file)
+    pipeline_id = os.path.split(source_folder)[-1]
+    logging.info('%s: created datapackage', pipeline_id)
 
     return package
 
 
-def bootstrap_pipeline(source_folder):
+def bootstrap_pipeline(package, source_folder):
```

Comment:
to answer your 1st question: for now yes. In the future all this can be refactored properly. I'm worried I'm micro-coding right now whereas my urgent goal is to get data through this bloody pipeline!